crates/wasi-common/src/snapshots/wasi_snapshot_preview1.rs

```diff
@@ -0,0 +1,1448 @@
+use crate::entry::{Descriptor, Entry};
+use crate::sandboxed_tty_writer::SandboxedTTYWriter;
+use crate::wasi::wasi_snapshot_preview1::WasiSnapshotPreview1;
+use crate::wasi::{types, AsBytes, Errno, Result};
+use crate::WasiCtx;
+use crate::{clock, fd, path, poll};
+use log::{debug, error, trace};
+use std::cell::Ref;
+use std::convert::TryInto;
+use std::io::{self, Read, Seek, SeekFrom, Write};
+use std::ops::DerefMut;
+use wiggle_runtime::{GuestBorrows, GuestPtr};
+
+impl<'a> WasiSnapshotPreview1 for WasiCtx {
```

Comment:
Excellent question! So I think it won't be that difficult. However, I suggest to leave that for a subsequent PR. I'm purposely migrated all impls into one module and directly into the trait impl so that we can have a better idea how to tackle the problem of multiple snapshots/polyfill. Does it make sense?

Reply:
I'd actually imagine that it makes it *easier* to share a `WasiCtx` between versions because we could implement both traits for one struct!